src/main/java/com/mraof/minestuck/effects/CreativeShockEffect.java

```diff
@@ -0,0 +1,166 @@
+package com.mraof.minestuck.effects;
+
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.item.artifact.CruxiteArtifactItem;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.network.StopCreativeShockEffectPacket;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ChorusFruitItem;
+import net.minecraft.item.EnderPearlItem;
+import net.minecraft.potion.Effect;
+import net.minecraft.potion.EffectType;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.event.entity.player.PlayerEvent;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
+import net.minecraftforge.event.world.ExplosionEvent;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+/**
+ * This is an adapted version of Cibernet's code in Minestuck Universe, credit goes to him!
+ * The different levels of amplification increase the range of the effect, allowing the selective limitation of both creative and non creative players.
+ * 0 = cant directly cause block breakage/placement outside of creative, 1 = cant access redstone machinery gui outside of creative, 2 = cant use mobility items outside of creative,
+ * 3 = cant directly cause block breakage/placement even in creative, 4 = cant access redstone machinery gui even in creative, 5 = cant use mobility items even in creative
```

Comment:
Idea: Define these different limitation levels as constants.
 For example a constant such as:
```java
public static final int LIMIT_MOBILITY_ITEMS = 2;
```
Used as:
```java
if(CreativeShockEffect.doesCreativeShockLimit(player, CreativeShockEffect.LIMIT_MOBILITY_ITEMS))
	*prevent elytra flying*
```